src/base/http/requesthandler.h

```diff
@@ -28,16 +28,21 @@
 
 #pragma once
 
+#include "iconnectionhandler.h"
+
 namespace Http
 {
     struct Environment;
     struct Request;
     struct Response;
 
-    class IRequestHandler
+    class RequestHandler : public IConnectionHandler
     {
     public:
-        virtual ~IRequestHandler() {}
+        virtual ~RequestHandler() = default;
         virtual Response processRequest(const Request &request, const Environment &env) = 0;
+
+    private:
+        void handleConnection(Connection *socket) override final;
```

Comment:
`final` implies `override` so it's redundant to use both.

Reply:
Maybe `Connection *connection`?

Reply:
And what's the point to redeclare it private? It implements interface method that's public by definition.

Reply:
>  And what's the point to redeclare it private? It implements interface method that's public by definition.

At some point in time, I was instructed to do something similar in another PR.

Reply:
>>  And what's the point to redeclare it private? It implements interface method that's public by definition.
>
>At some point in time, I was instructed to do something similar in another PR.

Are you sure it was about exactly the same case? Maybe it was about the case when you override previously protected method in final class?

Reply:
>final implies override so it's redundant to use both.

No it doesn't: https://stackoverflow.com/questions/29412412/does-final-imply-override

Reply:
> No it doesn't: https://stackoverflow.com/questions/29412412/does-final-imply-override

In such case it does (defacto). `final` requires the method to be `virtual`. But since this declaration does not have `virtual` specifier, it can only be _virtual_ if it is _overriden_. I would prefer to use minimal set of specifiers in method declarations. Or does it seem less readable to you?

Reply:
I was taking your comment at face value and didn't read the code. But if that particular function isn't virtual then it cannot be `override` and/or `final` anyway. It would be a compile error. Or are you saying something else?

Reply:
Scratch whatever I said. I missed your point and the rest of the SO answer. Yes, `final override` is superfluous. Because `final` required a virtual function aka something that is overridable (sic). In that case `override` won't achieve something more. Maybe only a clearer intent indication to the reader. But let's not go there.